mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```
@@ -55,86 +56,168 @@ static cl::opt<std::string> outputFilename("o", cl::desc("Output filename"),
+    PassManager pm(module.getContext(), PassManager::Nesting::Implicit);
+    pm.addPass(tosa::createTosaPartitionPass());
+    pm.addPass(miopen::createMIOpenCloneKernelsPass());
+
+    if (failed(pm.run(module))) {
+      return failure();
+    }
+  }
+
+  // Find kernel module, defaults to top module
+  llvm::StringRef modName = "__miopen";
+  auto kernelModule = module.lookupSymbol<ModuleOp>(modName);
+  if (!kernelModule) {
+    kernelModule = module;
+  }
+
+  PassManager pm(kernelModule.getContext(), PassManager::Nesting::Implicit);
   applyPassManagerCLOptions(pm);
 
-  // Set up high-level pipeline.
-  bool isHighLevel = highLevelPipeline.getValue();
+  bool isHighLevel = hostPipelineSet.contains("highlevel");
   if (isHighLevel) {
     miopen::addHighLevelPipeline(pm);
   }
 
   // Set up lowering pipeline.
-  if (miopenBuiltinPipeline.getValue()) {
-    StringRef pipeline = loweringTargetDialect.getValue();
-    if (pipeline == "tuning") {
-      // Set up the default lowering pipeline which goes down to affix tuning
-      // parameters
-      populateTuningPipeline(pm);
-    } else if (pipeline == "gpu") {
-      // Set up the default lowering pipeline which goes down to GPU dialect.
-      miopen::addPipeline(pm);
-    } else if (pipeline == "rocdl") {
-      // Set up the lowering pipeline which goes down to ROCDL dialect.
-      miopen::addPipeline(pm);
-      pm.addPass(createLowerGpuOpsToROCDLOpsPass(/*indexBitWidth=*/32));
+  if (kernelPipelineSet.contains("binary")) {
+    // Set up the lowering pipeline which goes down to ROCDL dialect.
+    int optLevel = gpuOpt.getValue();
+    if (optLevel < 0 || optLevel > 3) {
+      llvm::errs() << "Invalid GPU optimization level: " << optLevel << "\n";
+      return failure();
     }
+    bool systemOverride = false;
+    if (tripleName.empty() && targetChip.empty() && features.empty()) {
+      systemOverride = true;
+    }
+    BackendUtils utils(tripleName, targetChip, features, systemOverride);
```

This has to stay in the `mlir-rocm-runner`. Instead, you need to force user to give all those values from command line.

If not, this will just compile silently use whatever machine you are on and give the binary with its arch to populate in the graph.

---

This is only needed if you compile to binary. As part of lowering to XMIR we need the binaries to plug into the host code in the target spec (see example below).

I can change this to error if the target chip is not specified.

---

Superseded by later discussions, marking this as resolved.